src/Microsoft.AspNet.Mvc.TagHelpers/TagHelperContentExtensions.cs

```diff
@@ -51,7 +52,18 @@ public static class TagHelperContentExtensions
 
```

Comment:
Why yes, it's purpose is ugly. That's one of the reasons I didn't want to make the method `public` when I wrote the original version.

Might be possible to avoid duplicating the ugliness if the `<link>` and `<script>` tag helpers created the additional elements using a `TagHelperOutput` instance per added element. That's possible with the new Razor code but may also be ugly.